zerver/forms.py

```diff
@@ -80,12 +80,13 @@ def check_subdomain_available(subdomain: str, from_management_command: bool=Fals
         raise ValidationError(error_strings['extremal dash'])
     if not re.match('^[a-z0-9-]*$', subdomain):
         raise ValidationError(error_strings['bad character'])
-    if from_management_command:
-        return
     if len(subdomain) < 3:
         raise ValidationError(error_strings['too short'])
-    if is_reserved_subdomain(subdomain) or \
-       Realm.objects.filter(string_id=subdomain).exists():
+    if Realm.objects.filter(string_id=subdomain).exists():
+        raise ValidationError(error_strings['unavailable'])
+    if allow_reserved_subdomain:
+        return
+    if is_reserved_subdomain(subdomain):
```

Comment:
This looks a lot better!

As a tiny point, it's good to think about potential future-proofing here -- what happens if we have some other thing we want to prevent, and we just add it to the end of the function without noticing the `if ... return`?  `--allow-reserved-subdomain` will skip that new check as well, which may not be intended.

It may be good to defend against that by saying:
```
if is_reserved_subdomain(subdomain) and not allow_reserved_subdomain:
    raise ValidationError(error_strings['unavailable'])
```
...which has the same end effect right now as what you wrote, but bounds the effects of the flag much more tightly.

Reply:
Ok, I would make the change